### PR TITLE
Fixed a lot of knownConditionTrueFalse

### PR DESCRIPTION
**Pull request by SunBlack**
Fixed a lot of knownConditionTrueFalse hints from CppCheck 1.85

### CODE REVIEW

File: gpu/kinfu/tools/kinfu_app_sim.cpp [hunk 1]
```diff
@@ -1078,47 +1078,46 @@ struct KinFuApp
...
-	  SampledScopeTime fps(time_ms, i);
-	  //run kinfu algorithm
-	  if (integrate_colors_)
-	    has_image = kinfu_ (depth_device_, image_view_.colors_device_);
-	  else
-	    has_image = kinfu_ (depth_device_);                  
-	}
-	
-      }
+      /*if (1==0){ // live capture - probably doesn't work anymore, left in here for comparison:
```
Comment (SergioRAgostinho) [inline gpu/kinfu/tools/kinfu_app_sim.cpp]: Just erase all things you commented in this file. Worst case scenario we have version control to go back in history.

File: keypoints/src/narf_keypoint.cpp [hunk 2]
```diff
@@ -591,9 +591,9 @@ NarfKeypoint::calculateSparseInterestImage ()
     else
     {
       // Reduce number of neighbors to go through by filtering close by points with the same angle
-      bool do_neighbor_size_reduction = true;
+      /*bool do_neighbor_size_reduction = true;
       if (do_neighbor_size_reduction)
-      {
+      {*/
```
Comment (SergioRAgostinho) [inline keypoints/src/narf_keypoint.cpp]: Same comment as before.

File: simulation/tools/sim_viewer.cpp [hunk 3]
```diff
@@ -446,8 +446,8 @@ void capture (Eigen::Isometry3d pose_in, string point_cloud_fname)
   pcl::PointCloud<pcl::PointXYZRGB>::Ptr pc_out (new pcl::PointCloud<pcl::PointXYZRGB>);
   bool write_cloud = true;
   
-  if (write_cloud)
-  {
+  /*if (write_cloud)
+  {*
```
Comment (SergioRAgostinho) [inline simulation/tools/sim_viewer.cpp]: In this file as well.

File: gpu/kinfu/tools/kinfu_app_sim.cpp [hunk 4]
```diff
@@ -1078,47 +1078,46 @@ struct KinFuApp
...
+          else
+            has_image = kinfu_ (depth_device_);                  
+        }
+      }else{*/
+        //simulate:
+        cout << " color: " << integrate_colors_ << "\n"; // integrate_colors_ seems to be zero
+        depth_device_.upload (depth_sim.data, depth_sim.step, depth_sim.rows, depth_sim.cols);
+        if (integrate_colors_){
+            image_view_.colors_device_.upload (rgb24_sim.data, rgb24_sim.step, rgb24_sim.rows, rgb24_sim.cols);
+        }
```
[the remark this answers is not in the document]
Comment (SunBlack) [reply gpu/kinfu/tools/kinfu_app_sim.cpp]: Should I remove all dead code here (including code in `1==0` block)?

[the remark this answers is not in the document]
Comment (SunBlack) [reply gpu/kinfu/tools/kinfu_app_sim.cpp]: What should happens with this variable in general? It is initialized with `false` and can only be set to true with `toggleColorIntegration`. But before any usage of this variable, value will be reverted to `false`, so `toggleColorIntegration` has no effect.

So there exists two options: Remove `toggleColorIntegration` or remove `integrate_colors_=false;` line in `execute` function. What do you prefer?

[the remark this answers is not in the document]
Comment (SergioRAgostinho) [reply gpu/kinfu/tools/kinfu_app_sim.cpp]: Good point 😄  This is legacy code from sprints which was never properly finished, but somehow got merged into the master branch.

From the code comments it seems color integration was not working properly. Unfortunately I don't have the hardware to verify this. Before making this decision I would like to actually have someone try to test the state of this feature.

This topic alone can be the subject of a PR on it's own, so I say we do that and for this one we just cleanup. I'll resolve the comments pertaining this topic.

File: gpu/people/tools/people_pcd_prob.cpp [hunk 5]
```diff
@@ -290,7 +290,7 @@ int main(int argc, char** argv)
   parse_argument (argc, argv, "-prob", saveProb);
 
   bool debugOutput = 0;
-  parse_argument (argc, argv, "-debug", saveProb);
+  parse_argument (argc, argv, "-debug", debugOutput);
```
[the remark this answers is not in the document]
Comment (SunBlack) [reply gpu/people/tools/people_pcd_prob.cpp]: See #2656

[the remark this answers is not in the document]
Comment (SunBlack) [reply gpu/people/tools/people_pcd_prob.cpp]: I removed this from this PR now.

File: simulation/tools/sim_viewer.cpp (hunk 3 above)
Comment (SunBlack) [reply simulation/tools/sim_viewer.cpp]: Ok, I removed a lot of more disabled code of this file now, too.

**Comment by SunBlack** [comment]
> For the future, don't address code style issues. We have a PR ready which adds clang_format and we're going to do a full wipe in a single commit.

Sorry, because of bad indention code was really bad readable (8 spaces for a tab instead of 4), so I fixed indention in this file ;)

**Pull request merged**